src/object/obj_class.c

```diff
@@ -1020,6 +1020,112 @@ daos_oclass_fit_max(daos_oclass_id_t oc_id, int domain_nr, int target_nr,
 	return oc ? 0 : -DER_NONEXIST;
 }
 
+int
+dc_set_oclass(daos_handle_t coh, int domain_nr, int target_nr,
+	      daos_ofeat_t ofeats, daos_oclass_hints_t hints,
+	      daos_oclass_id_t *oc_id_p)
+{
+	uint64_t		rf_factor;
+	daos_oclass_id_t	cid = 0;
+	struct daos_obj_class	*oc;
+	struct daos_oclass_attr	ca;
+	uint16_t		shd, rdd;
+	int			grp_size;
+
+	rf_factor = dc_cont_hdl2redunfac(coh);
+	rdd = hints & DAOS_OCH_RDD_MASK;
+	shd = hints & DAOS_OCH_SHD_MASK;
+
+	/** first set a reasonable default based on RF & RDD hint (if set) */
+	switch (rf_factor) {
+	case DAOS_PROP_CO_REDUN_RF0:
+		if (rdd == DAOS_OCH_RDD_RP)
+			cid = OC_RP_2GX;
+		else if (rdd == DAOS_OCH_RDD_EC)
+			cid = OC_EC_2P1G1;
+		else
+			cid = OC_SX;
+		break;
+	case DAOS_PROP_CO_REDUN_RF1:
+		if (rdd == DAOS_OCH_RDD_RP)
+			cid = OC_RP_2GX;
+		else if (rdd == DAOS_OCH_RDD_EC || ofeats & DAOS_OF_ARRAY ||
+			 ofeats & DAOS_OF_ARRAY_BYTE)
+			/** TODO - this should be GX when supported */
+			cid = OC_EC_2P1G1;
+		else
+			cid = OC_RP_2GX;
+		break;
+	case DAOS_PROP_CO_REDUN_RF2:
+		if (rdd == DAOS_OCH_RDD_RP)
+			cid = OC_RP_3GX;
+		else if (rdd == DAOS_OCH_RDD_EC || ofeats & DAOS_OF_ARRAY ||
+			 ofeats & DAOS_OF_ARRAY_BYTE)
+			/** TODO - this should be GX when supported */
+			cid = OC_EC_2P2G1;
```

Comment:
must it be 2P1 or 2P2? is it possible to select 4P2, 8P2, 16P2 in future?

Reply:
yea, i got those from @wangdi1 
there are some that are not available today, and this is what was suggested as a reasonable default if not set by user.

Reply:
agreed that we probably need more correlation between the domain but also the target nr to set also the GX.
we can tackle this in a future PR when we add more EC object class?

Reply:
I've added all classes (commit 33406afa7c15ffc2d11e89b058da70b878e6404b), I assume we are not going to land it to 1.2, so we can switch to new classes in follow-on patch

Reply:
we actually want to add to 1.2.
let me see what you added and adjust this PR i guess.

Reply:
ah wait.. if we are not landing your PR to 1.2, then i can't change it here since we want to land this to 1.2.
i can do the change to master in a future PR to master. so if you are OK with, please land it